src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
Ah. The problem is that you're passing a shared_ptr, not the cLuaWorldTask. And looking into it, fixing this is more complicated because of cResettable is a non-copyable type. A simple workaround is to use lambda for the bind, which is probably clearer.